web/src/frontend/classic/stores/Documentation.ts

```diff
@@ -0,0 +1,80 @@
+import { writable } from 'svelte/store';
+
+export type DocCategory = {
+    name: string;
+    description: HTMLElement;
+    sections: DocSection[];
+}
+export type DocSection = {
+    name: string,
+    content: HTMLElement
+}
+
+export const documentation = writable<DocCategory[]>([]);
+
+/**
+ * Creates a slide section by extracting HTML elements from the given content element until a specified split element is encountered.
+ * @param content - The content element from which to extract the slide section.
+ * @param split - The element name at which to stop extracting and split the slide section.
+ * @returns The created slide section element.
+ */
+function slideElement(content:HTMLElement, split:string[]) : HTMLElement {
+    const slide = document.createElement("div");
+    let nextSibling = content.nextSibling;
+    while (nextSibling && !split.includes(nextSibling.nodeName)) {
+        slide.appendChild(nextSibling.cloneNode(true));
+        nextSibling = nextSibling.nextSibling;
+    }
+    return slide;
+}
+
+(async function () {
+    const url = 'https://hakuneko.download/docs/haruneko/';
+    // TODO Documentation should be cached using a service worker
+    const cacheKey = `cache:${url}`;
```

Comment:
Should the search parameters and fragment of the `url` be included when using it as key for caching?

Reply:
not it this case but the pattern was easy to reuse. I was first  trying to cache images as well.